worldedit-core/src/main/java/com/sk89q/worldedit/math/convolution/SnowHeightMap.java

```diff
@@ -0,0 +1,218 @@
+                double scale = (double) (curHeight - originY) / (double) (newHeight - originY);
+
+                // Depending on growing or shrinking we need to start at the bottom or top
+                if (newHeight >= curHeight) {
+                    // Set the top block of the column to be the same type (this might go wrong with rounding)
+                    BlockState existing = session.getBlock(BlockVector3.at(xr, curHeight, zr));
+
+                    // Skip water/lava
+                    if (!existing.getBlockType().getMaterial().isLiquid()) {
+                        setSnowLayer(xr, zr, newHeight, layers);
+                        ++blocksChanged;
+
+                        // Grow -- start from 1 below top replacing airblocks
+                        for (int y = (int) newHeight - 1 - originY; y >= 0; --y) {
+                            if (y >= newHeight - 1 - originY - layerBlocks) {
+                                session.setBlock(BlockVector3.at(xr, originY + y, zr), fillerSnow);
+                            } else {
+                                int copyFrom = (int) (y * scale);
+                                session.setBlock(BlockVector3.at(xr, originY + y, zr),
+                                        session.getBlock(BlockVector3.at(xr, originY + copyFrom, zr)));
+                            }
+                            ++blocksChanged;
+                        }
+                    }
+                } else {
+                    // Shrink -- start from bottom
+                    for (int y = 0; y < (int) newHeight - originY; ++y) {
+                        if (y >= (int) newHeight - originY - layerBlocks) {
+                            session.setBlock(BlockVector3.at(xr, originY + y, zr), fillerSnow);
+                        } else {
+                            int copyFrom = (int) (y * scale);
+                            session.setBlock(BlockVector3.at(xr, originY + y, zr),
+                                    session.getBlock(BlockVector3.at(xr, originY + copyFrom, zr)));
+                        }
+                        ++blocksChanged;
+                    }
+
+                    setSnowLayer(xr, zr, newHeight, layers);
+                    ++blocksChanged;
+
+                    // Fill rest with air
+                    for (int y = (int) newHeight + 1; y <= curHeight; ++y) {
+                        session.setBlock(BlockVector3.at(xr, y, zr), fillerAir);
+                        ++blocksChanged;
+                    }
+                }
+            }
+        }
+
+        // Drop trees to the floor -- TODO
+        return blocksChanged;
+    }
+
+    private void setSnowLayer(int x, int z, float newHeight, IntegerProperty property) throws MaxChangedBlocksException {
+        int layers = (int) ((newHeight % 1) * 8) + 1;
+        session.setBlock(BlockVector3.at(x, (int) newHeight, z), BlockTypes.SNOW.getDefaultState().with(property, layers));
```

Comment:
```suggestion
        session.setBlock(BlockVector3.at(x, (int) newHeight, z), BlockTypes.SNOW.getState(ImmutableMap.of(property, layers));
```
slightly more efficient :)